fractal/cmd/_dataset.py

```diff
@@ -22,16 +22,25 @@ async def post_dataset(
     client: AuthClient,
     project_id: int,
     dataset_name: str,
-    metadata_filename: Optional[str] = None,
+    metadata: Optional[str] = None,
     type: Optional[str] = None,
     batch: bool = False,
-    **kwargs,
 ) -> RichJsonInterface:
-
-    if metadata_filename is None:
+    """
```

Comment:
Our typical docstring format (especially in fractal-server) is:
```python
def _check_credentials(
    *, username: Union[str, None], password: Union[str, None]
) -> Tuple[str, str]:
    """
    Check that username and password are defined

    Arguments:
        username: Username
        password: Password
    """
```

(no need for type hints, since they are already in the function signature)

Reply:
Hmm, in the task functions I looked at (cellpose, create ome zarr), we use this notation:
```
     :param diameter: Expected object diameter in pixels for cellpose
```

Are we changing our docstring style?

Reply:
Tasks and server/client have two different styles (and it's unfortunate, I totally agree).

At least in the fractal-server docs we make it explicit:
> The documentations is built with [mkdocs](https://www.mkdocs.org/) and the [Material theme](https://squidfunk.github.io/mkdocs-material). Whenever possible, docstrings should be formatted as in the [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings).

For the tasks, we implicitly use https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html#the-sphinx-docstring-format, with minor changes (e.g. we infer the type hints directly from the call signature)

As for the client, we do not produce documentation and then this issue is less crucial, but I'd say we stay closer to the fractal-server style.